exec/plugin-commands-script-runners/src/run.ts

```diff
@@ -300,6 +304,44 @@ ${renderCommands(rootScripts)}`
   return output
 }
 
+export interface RunScriptOptions {
+  enablePrePostScripts: boolean
+}
+
+export const runScript: (opts: {
+  manifest: ProjectManifest
+  lifecycleOpts: RunLifecycleHookOptions
+  runScriptOptions: RunScriptOptions
+  passedThruArgs: string[]
+}, scriptName: string) => Promise<void> = async function (opts, scriptName) {
+  if (
+    opts.runScriptOptions.enablePrePostScripts &&
+    opts.manifest.scripts?.[`pre${scriptName}`] &&
+    !opts.manifest.scripts[scriptName].includes(`pre${scriptName}`)
+  ) {
+    await runLifecycleHook(`pre${scriptName}`, opts.manifest, opts.lifecycleOpts)
+  }
+  await runLifecycleHook(scriptName, opts.manifest, { ...opts.lifecycleOpts, args: opts.passedThruArgs })
+  if (
+    opts.runScriptOptions.enablePrePostScripts &&
+    opts.manifest.scripts?.[`post${scriptName}`] &&
+    !opts.manifest.scripts[scriptName].includes(`post${scriptName}`)
+  ) {
+    await runLifecycleHook(`post${scriptName}`, opts.manifest, opts.lifecycleOpts)
+  }
+}
+
 function renderCommands (commands: string[][]) {
   return commands.map(([scriptName, script]) => `  ${scriptName}\n    ${script}`).join('\n')
 }
+
+function getSpecifiedScripts (scripts: PackageScripts, scriptName: string) {
+  const specifiedSelector = getSpecifiedScripts(scripts, scriptName)
```

Comment:
CI failed.

<img width="1409" alt="image" src="https://user-images.githubusercontent.com/8198408/215383691-27c4cb4e-000d-4e4c-970a-412a6a8f4044.png">

https://github.com/pnpm/pnpm/actions/runs/4037841963/jobs/6941420302#step:10:2436

Reply:
Sorry 🙏 I fixed the code to fix CI error